libafl_bolts/src/shmem.rs

```diff
@@ -877,9 +935,10 @@ pub mod unix_shmem {
                     let map = shmat(id_int, ptr::null(), 0) as *mut c_uchar;
 
                     if map.is_null() || map == ptr::null_mut::<c_uchar>().wrapping_sub(1) {
-                        return Err(Error::unknown(
-                            "Failed to map the shared mapping".to_string(),
-                        ));
+                        perror(b"shmat\0".as_ptr() as *const _);
```

Comment:
There is `std::io::Error::last_os_error().raw_os_error().unwrap()` which gives us the errno string if I read this correctly, so that could be used for `Error` here